reference/FungibleToken.scilla

```diff
@@ -294,105 +269,171 @@ transition AuthorizeOperator(operator: ByStr20)
+  is_operator_approved <- exists operators_map[from][_sender];
+  is_default_operator = is_default_operator f_eq _sender default_operators;
+  is_revoked_operator <- exists revoked_default_operators[from][_sender];
+  is_default_operator_approved = let is_not_revoked_operator = 
+    negb is_revoked_operator in andb is_not_revoked_operator is_default_operator;
+  is_approved = orb is_operator_approved is_default_operator_approved;
+  match is_approved with
+    | False =>
+      err = CodeNotAuthorized;
+      issueErrorEvent err
+    | True =>
+      authorizedMoveIfSufficientBalance from to amount;
+      e = {_eventname : "OperatorSendSuccess"; initiator : _sender; sender : from; recipient : to; amount : amount};
+      event e;
+      (* Prevent sending to a contract address that does not support transfers of token *)
+      msg_to_recipient = {_tag : "recipientAcceptOperatorSend"; _recipient : to; _amount : zero; 
+                          initiator : _sender; sender : from; recipient : to; amount : amount};
+      msg_to_sender = {_tag : "operatorSendSuccessCallBack"; _recipient : _sender; _amount : zero; 
+                      initiator : _sender; sender : from; recipient : to; amount : amount};
+      msgs = two_msgs msg_to_recipient msg_to_sender;
+      send msgs
+   end
 end
 
-(* @dev: Emits an event with the amount of tokens owned by an address. *)
-transition BalanceOf(address: ByStr20)
-  get_bal <- balances_map[address];
-  balance = get_balance get_bal;
-  e = {_eventname: "BalanceOf"; balance: balance};
-  event e
+(* @dev: Move a given amount of tokens from one address to another using the allowance mechanism. The caller must be an approved_spender. *)
+(* @dev: Balance of recipient will increase. Balance of token_owner will decrease.                                                        *)
+(* @param from:    Address of the token_owner whose balance is decreased.                                                                 *)
+(* @param to:      Address of the recipient whose balance is increased.                                                                   *)
+(* @param amount:  Amount of tokens to be transferred.                                                                                    *)
+transition transferFrom(from: ByStr20, to: ByStr20, amount: Uint128)
+  get_bal <- balances_map[from];
+  match get_bal with
+  | None => 
+    err = CodeNotFound;
+    issueErrorEvent err
+  | Some bal =>
+    get_spender_allowed <- allowances_map[from][_sender];
+    match get_spender_allowed with
+    | None =>
+      err = CodeNotAuthorized;
+      issueErrorEvent err
+    | Some allowed =>
+      min = min_int bal allowed;
+      can_do = uint128_le amount min;
+      match can_do with
+      | True =>
+        authorizedMoveIfSufficientBalance from to amount;
+        e = {_eventname : "TransferFromSuccess"; initiator : _sender; sender : from; recipient : to; amount : amount};
```

Comment:
Should we check the order of the files in the event? It does not match the signature that XSGD has.

Reply:
I think the logical order of the events should be:
1. `initiator`
2. `sender`
3. `recipient`
4. `amount`
This should make more sense to the app developer, as they can understand the logical order of who did what. Should we deviate from XSGD for now for this? I feel it is a better order.